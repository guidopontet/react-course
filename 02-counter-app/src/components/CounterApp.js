import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {

  const [ counter, setCounter ] = useState(value);

  const addNumber = () => {
    setCounter(counter + 1);
  };

  const lessNumber = () => {
    setCounter(counter - 1);
  }

  const resetNumber = () => {
    setCounter(value);
  }


  return (
    <Fragment>
      <h1>CounterApp</h1>
      <p>{ counter }</p>

      <button onClick={ lessNumber }>-1</button>
      <button onClick={ resetNumber }>Reset</button>
      <button onClick={ addNumber }>+1</button>
    </Fragment>
  );
};

CounterApp.prototypes = {
  value: PropTypes.number,
};

CounterApp.defaultProps = {
  value: 1,
};

export default CounterApp;
