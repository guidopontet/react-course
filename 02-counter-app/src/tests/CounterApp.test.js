import "@testing-library/jest-dom";

import { shallow } from 'enzyme';

import CounterApp from "../components/CounterApp";

describe('CounterApp', () => {
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test('should show CounterApp', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should show default', () => {
    const wrapper = shallow(<CounterApp value={ 100 }/>);

    const counterText = wrapper.find('p').text().trim();

    expect(counterText).toBe("100");
  });

  test('should add counter 1 unit', () => {
    wrapper.find('button').at(2).simulate('click');

    const counterText = wrapper.find('p').text().trim();

    expect(counterText).toBe("11");
  });

  test('should minus counter 1 unit', () => {
    wrapper.find('button').at(0).simulate('click');

    const counterText = wrapper.find('p').text().trim();

    expect(counterText).toBe("9");
  });

  test('should reset counter', () => {
    const value = '5';
    const wrapper = shallow(<CounterApp value={ value }/>);

    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');

    const counterText = wrapper.find('p').text().trim();

    expect(counterText).toBe(value);
  });
 });
