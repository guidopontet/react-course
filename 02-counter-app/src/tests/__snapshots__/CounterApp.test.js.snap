// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp should show CounterApp 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <p>
    10
  </p>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;
